tbd/template/template.go

```diff
@@ -0,0 +1,64 @@
+package template
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"path/filepath"
+)
+
+type MissingError string
+
+func (e MissingError) Error() string {
+	return fmt.Sprintf("no template exists with name '%s'", string(e))
+}
+
+type Templates map[string]*template.Template
+
+type Template func(io.Writer, interface{}) error
```

Comment:
The only downside of this is that the mocks become slightly more annoying to use